src/Mod/Arch/ArchRoof.py

```diff
@@ -42,241 +52,328 @@ def QT_TRANSLATE_NOOP(ctxt,txt):
                 if "Face" in sel.SubElementNames[0]:
-                    idx = int(sel.SubElementNames[0][4:])
-                    FreeCAD.ActiveDocument.openTransaction(translate("Arch","Create Roof"))
+                    i = int(sel.SubElementNames[0][4:])
+                    FreeCAD.ActiveDocument.openTransaction(translate("Arch", "Create Roof"))
                     FreeCADGui.addModule("Arch")
-                    FreeCADGui.doCommand("obj = Arch.makeRoof(FreeCAD.ActiveDocument."+obj.Name+","+str(idx)+")")
+                    FreeCADGui.doCommand("obj = Arch.makeRoof(FreeCAD.ActiveDocument." + obj.Name + "," + str(i) + ")")
                     FreeCADGui.addModule("Draft")
                     FreeCADGui.doCommand("Draft.autogroup(obj)")
                     FreeCAD.ActiveDocument.commitTransaction()
                     FreeCAD.ActiveDocument.recompute()
                     return
-            if hasattr(obj,'Shape'):
+            if hasattr(obj, "Shape"):
                 if obj.Shape.Wires:
-                    FreeCAD.ActiveDocument.openTransaction(translate("Arch","Create Roof"))
+                    FreeCAD.ActiveDocument.openTransaction(translate("Arch", "Create Roof"))
                     FreeCADGui.addModule("Arch")
-                    FreeCADGui.doCommand("obj = Arch.makeRoof(FreeCAD.ActiveDocument."+obj.Name+")")
+                    FreeCADGui.doCommand("obj = Arch.makeRoof(FreeCAD.ActiveDocument." + obj.Name + ")")
                     FreeCADGui.addModule("Draft")
                     FreeCADGui.doCommand("Draft.autogroup(obj)")
                     FreeCAD.ActiveDocument.commitTransaction()
                     FreeCAD.ActiveDocument.recompute()
                     return
             else:
-                FreeCAD.Console.PrintMessage(translate("Arch","Unable to create a roof"))
+                FreeCAD.Console.PrintMessage(translate("Arch", "Unable to create a roof"))
         else:
-            FreeCAD.Console.PrintMessage(translate("Arch","Please select a base object")+"\n")
+            FreeCAD.Console.PrintMessage(translate("Arch", "Please select a base object") + "\n")
             FreeCADGui.Control.showDialog(ArchComponent.SelectionTaskPanel())
-            FreeCAD.ArchObserver = ArchComponent.ArchSelectionObserver(nextCommand="Arch_Roof")
+            FreeCAD.ArchObserver = ArchComponent.ArchSelectionObserver(nextCommand = "Arch_Roof")
             FreeCADGui.Selection.addObserver(FreeCAD.ArchObserver)
 
 
 class _Roof(ArchComponent.Component):
-
-    "The Roof object"
-
-    def __init__(self,obj):
-
-        ArchComponent.Component.__init__(self,obj)
+    '''The Roof object'''
+    def __init__(self, obj):
+        ArchComponent.Component.__init__(self, obj)
         self.setProperties(obj)
         obj.IfcType = "Roof"
         obj.Proxy = self
 
-    def setProperties(self,obj):
-
+    def setProperties(self, obj):
         pl = obj.PropertiesList
         if not "Angles" in pl:
```

Comment:
Since you are cleaning up the properties, you can also improve this
```py
if "Angles" not in pl:
```

Reply:
Cosmetics, to be handled later.